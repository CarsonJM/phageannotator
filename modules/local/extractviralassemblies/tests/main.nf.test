nextflow_process {

    name "Test process: EXTRACTVIRALASSEMBLIES"
    script "../main.nf"
    process "EXTRACTVIRALASSEMBLIES"
    tag "modules"
    tag "modules_local"
    tag "extractviralassemblies"


    test("fasta.gz & 'NODE_1_length_20973_cov_191.628754'") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                        file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz", checkifExists: true),
                ]
                input[1] = "NODE_1_length_20973_cov_191.628754"
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz & 'NODE_1_length_20973_cov_191.628754' - stub") {

        options "-stub"

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                        file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz", checkifExists: true),
                ]
                input[1] = "NODE_1_"
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
