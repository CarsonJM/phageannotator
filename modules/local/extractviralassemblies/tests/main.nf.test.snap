{
    "fasta.gz & 'NODE_1_length_20973_cov_191.628754'": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_viral_assemblies.tsv:md5,d98e022dc9c925d607bbbce2a5e62055"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8a8d145ac2dc055ca6f09a8aed8c0010"
                ],
                "versions": [
                    "versions.yml:md5,8a8d145ac2dc055ca6f09a8aed8c0010"
                ],
                "viral_assemblies": [
                    [
                        {
                            "id": "test"
                        },
                        "test_viral_assemblies.tsv:md5,d98e022dc9c925d607bbbce2a5e62055"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-28T12:19:51.567033391"
    },
    "fasta.gz & 'NODE_1_length_20973_cov_191.628754' - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_viral_assemblies.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8a8d145ac2dc055ca6f09a8aed8c0010"
                ],
                "versions": [
                    "versions.yml:md5,8a8d145ac2dc055ca6f09a8aed8c0010"
                ],
                "viral_assemblies": [
                    [
                        {
                            "id": "test"
                        },
                        "test_viral_assemblies.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-28T12:20:09.552372442"
    }
}